addons/l10n_it_edi/models/account_edi_format.py

```diff
@@ -228,41 +228,55 @@ def _l10n_it_goods_in_italy(self, invoice):
         return bool(invoice_lines_tags & vj3_lines_tags)
 
     def _l10n_it_document_type_mapping(self):
+        """ Returns a dictionary with the required features for every TDxx FatturaPA document type """
         return {
-            'TD01': dict(move_types=['out_invoice'], import_type='in_invoice', downpayment=False),
-            'TD02': dict(move_types=['out_invoice'], import_type='in_invoice', downpayment=True),
-            'TD04': dict(move_types=['out_refund'], import_type='in_refund'),
-            'TD07': dict(move_types=['out_invoice'], import_type='in_invoice', simplified=True),
-            'TD08': dict(move_types=['out_refund'], import_type='in_refund', simplified=True),
-            'TD09': dict(move_types=['out_invoice'], import_type='in_invoice', simplified=True),
-            'TD17': dict(move_types=['in_invoice', 'in_refund'], import_type='in_invoice', self_invoice=True, services_or_goods="service"),
-            'TD18': dict(move_types=['in_invoice', 'in_refund'], import_type='in_invoice', self_invoice=True, services_or_goods="consu", partner_in_eu=True),
-            'TD19': dict(move_types=['in_invoice', 'in_refund'], import_type='in_invoice', self_invoice=True, services_or_goods="consu", goods_in_italy=True),
+            'TD01': dict(move_types=['out_invoice'], import_type='in_invoice', self_invoice=False, simplified=False, downpayment=False),
+            'TD02': dict(move_types=['out_invoice'], import_type='in_invoice', self_invoice=False, simplified=False, downpayment=True),
+            'TD04': dict(move_types=['out_refund'], import_type='in_refund', self_invoice=False, simplified=False),
+            'TD07': dict(move_types=['out_invoice'], import_type='in_invoice', self_invoice=False, simplified=True),
+            'TD08': dict(move_types=['out_refund'], import_type='in_refund', self_invoice=False, simplified=True),
+            'TD09': dict(move_types=['out_invoice'], import_type='in_invoice', self_invoice=False, simplified=True),
+            'TD17': dict(move_types=['in_invoice', 'in_refund'], import_type='in_invoice', simplified=False, self_invoice=True, services_or_goods="service"),
+            'TD18': dict(move_types=['in_invoice', 'in_refund'], import_type='in_invoice', simplified=False, self_invoice=True, services_or_goods="consu", goods_in_italy=False, partner_in_eu=True),
+            'TD19': dict(move_types=['in_invoice', 'in_refund'], import_type='in_invoice', simplified=False, self_invoice=True, services_or_goods="consu", goods_in_italy=True),
+            'TD28': dict(move_types=['in_invoice', 'in_refund'], import_type='in_invoice', simplified=False, self_invoice=True, partner_country_code="SM")
         }
 
-    def _l10n_it_get_document_type(self, invoice):
-        is_simplified = self._l10n_it_edi_is_simplified(invoice)
-        is_self_invoice = self._l10n_it_edi_is_self_invoice(invoice)
+    def _l10n_it_get_invoice_features_for_document_type_selection(self, invoice):
```

Comment:
Now to select the document type we get a dictionary of "features" from the to be compared with those of the Document Type. When they are equals, the Document Type is selected